### ISSUE TITLE
UnexpectedElasticsearchClientException when using Elasticsearch.net 5.x

### ISSUE THREAD

**Issue by odyth** [DataDog/dd-trace-dotnet#253]
I installed trace agent 0.7 to test it out. After installing every call to ElasticSearch fails. After uninstalling the trace agent things returned to normal. Something that is being injected into the code causes a cast exception. All the calls to elastic search are async. This is a C# .net 4.7.2 app running in IIS on windows 2016. I am using Elasticsearch.net version 5.6.1 and Nest 5.6.1 to make my outbound api calls.

I would see different type of UnexpectedElasticsearchClientException in my logs

`Unable to cast object of type 'System.Threading.Tasks.Task`1[Elasticsearch.Net.ElasticsearchResponse`1[Nest.SearchResponse`1[MyNameSpace.MyClass]]]' to type 'System.Threading.Tasks.Task`1[Nest.SearchResponse`1[MyNameSpace.MyClass]]'.`

and 

`Unable to cast object of type 'System.Threading.Tasks.Task`1[Elasticsearch.Net.ElasticsearchResponse`1[Nest.MultiGetResponse]]' to type 'System.Threading.Tasks.Task`1[Nest.MultiGetResponse]'.
`

Both have the same stack trace.

Stack trace:
```
at Elasticsearch.Net.Transport`1.<RequestAsync>d__15`1.MoveNext() in c:\Projects\elastic\net-5\src\Elasticsearch.Net\Transport\Transport.cs:line 151
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Nest.ElasticClient.<Nest-IHighLevelToLowLevelDispatcher-DispatchAsync>d__261`4.MoveNext() in c:\Projects\elastic\net-5\src\Nest\ElasticClient.cs:line 76
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
```

If I log into datadog and look at the trace with the error I see the above exceptions and this is all that is recorded in the stack in datadog.
```
 at CallElasticsearchAsync(Object , Object , CancellationToken )
   at Datadog.Trace.ClrProfiler.Integrations.ElasticsearchNetIntegration.<CallElasticsearchAsyncInternal>d__15`1.MoveNext()
```

**Comment by odyth** [comment]
I went back and looked at the documentation and see you have listed support for version 6.0+ of NEST / Elasticsearch.Net. I was expecting to just not get elastic search tracing since I was using 5.x....instead of having it cause every request to throw a cast exception because my library is a bit older....

Really feel like the appropriate action is to detect incompatibility and disable tracing.

**Comment by lucaspimentel** [comment]
Hey, @odyth. Disabling tracing automatically when we detect incompatible libraries is on our roadmap. We will also add  support for tracing `NEST` 5.x sometime after the 1.0 release.

In the meantime, you can disable tracing for Elasticsearch manually by adding the following environment variable:
```
DD_DISABLED_INTEGRATIONS=ElasticsearchNet
```

**Comment by odyth** [comment]
Ok thanks for the update. I had one more issue with the tracer that I am having a hard time explaining. One of my API calls gets 100% error rate in datadog and the stack trace when you look at the trace in datadog is below. The only thing that is different about this call is the content would be null (it's a mock endpoint for the moment). The method returns a HttpResponseMessage and the content would be null with a status code of 204 no content.

```
at System.Web.Http.Controllers.ApiControllerActionSelector.ActionSelectorCacheItem.SelectAction(HttpControllerContext controllerContext)
   at System.Web.Http.ApiController.ExecuteAsync(HttpControllerContext controllerContext, CancellationToken cancellationToken)
   at Datadog.Trace.ClrProfiler.Integrations.AspNetWebApi2Integration.<ExecuteAsyncInternal>d__2.MoveNext()
```

**Comment by lucaspimentel** [comment]
@odyth: can you create a separate issue in github to track that Web API error? We can keep this one open to track the fix for `NEST` 5.x.

**Comment by lucaspimentel** [comment]
We are adding stricter validation of instrumented assemblies to resolve this issue. PRs https://github.com/DataDog/dd-trace-dotnet/pull/339 and https://github.com/DataDog/dd-trace-dotnet/pull/341 make it so we only instrument `Elasticsearch.Net` 6.x, so the exception with 5.x no longer happens.

PR https://github.com/DataDog/dd-trace-dotnet/pull/348 adds support for instrumenting `Elasticsearch.Net` 5.x as a separate integration.

These changes will be included in release 1.2.

**Comment by lucaspimentel** [comment]
[Release 1.2](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.2.0) is out and adds support for `Elasticsearch.Net` 5.x.
